tests/units/scripts/test_run_tsfresh.py

```diff
@@ -31,6 +32,7 @@ def extract_features_mock(df, **kwargs):
 
     def tearDown(self):
         # Remove the directory after the test
+        os.chdir(self.curr_dir)
```

Comment:
what is the motivation behind the new 
`os.chdir`?

Reply:
Did you try this with xdist? (the parallel test runner plugin for pytest)

I am not sure what happens then

Reply:
I have not tried xdist. From quick googling around I'm not sure I understand how to make it work. Could you please let me know how, and I will be happy to test? Thanks

Reply:
But also notice in the `setUp()` method in the same `test_run_tsfresh.py` file where I'm already setting the `self.curr_dir` property to the working directory of the underlying process. I don't believe this is controversial at all.

Reply:
xdist is the parallelization plugin of pytest. You can install it by `pip install pytest-xdist`. 

I was worried that only the mainprocess calls `setUp()` and that this would create problems if the child processes call `tearDown`. However, it seems that each process calls these methods individually (see e.g. https://github.com/pytest-dev/pytest-xdist/issues/15)

So, from the xdist point of view this change is fine